lang/LangPrimSource/PyrSched.cpp

```diff
@@ -568,36 +500,7 @@ SCLANG_DLLEXPORT_C void schedRun() {
     SC_Thread thread(schedRunFunc);
     gSchedThread = std::move(thread);
 
-#ifdef __APPLE__
-    int policy;
-    struct sched_param param;
-
-    // pthread_t thread = pthread_self ();
-    pthread_getschedparam(gSchedThread.native_handle(), &policy, &param);
-    // post("param.sched_priority %d\n", param.sched_priority);
-
-    policy = SCHED_RR; // round-robin, AKA real-time scheduling
-
-    int machprio;
-    boolean_t timeshare;
-    GetStdThreadSchedule(pthread_mach_thread_np(gSchedThread.native_handle()), &machprio, &timeshare);
-    // post("mach priority %d   timeshare %d\n", machprio, timeshare);
-
-    // what priority should gSchedThread use?
-
-    RescheduleStdThread(pthread_mach_thread_np(gSchedThread.native_handle()), 62, false);
-
-    GetStdThreadSchedule(pthread_mach_thread_np(gSchedThread.native_handle()), &machprio, &timeshare);
-    // post("mach priority %d   timeshare %d\n", machprio, timeshare);
-
-    // param.sched_priority = 70; // you'll have to play with this to see what it does
-    // pthread_setschedparam (gSchedThread, policy, &param);
-
-    pthread_getschedparam(gSchedThread.native_handle(), &policy, &param);
-
-    // post("param.sched_priority %d\n", param.sched_priority);
-#endif // __APPLE__
-
+    // NOTE: on macOS, we set the thread priority in schedRunFunc using the QoS API.
```

Comment:
Why not also on Linux (with `pthread_self()`)? IMO it would be better if the thread priority code were in the same place.